### ISSUE TITLE
Move video between instances

### ISSUE THREAD

**Issue by yassinphilip** [Chocobozzz/PeerTube#549]
Is there a way to move a video from instance A to instance B, re-encode it on instance B, delete it frome instance A and keep to page views tally count?

**Comment by Chocobozzz** [comment]
What you're looking for is the account migration feature. It's not implemented sorry.

**Comment by Deuchnord** [comment]
I would also be interested in a way to migrate user data from an instance to another, as I have planned to open my own instance in the future :slightly_smiling_face:

**Comment by FinlayDaG33k** [comment]
Has there been any updates on this matter?

**Comment by rigelk** [comment]
@FinlayDaG33k no update on the matter, no. Otherwise the issue would have been updated accordingly ;)

As for the issue documentation, maybe some interesting read under the viewpoint of backup/restore procedures: https://the-federation.info/specs/backup-restore/

**Comment by DontUseGithub** [comment]
relevant https://www.reddit.com/r/PeerTube/comments/ekmjeb/peertubevideo_one_of_the_largest_instances_is/

**Comment by rigelk** [comment]
Before we tackle any form of automatic account migration, we could implement simple import/export of:
- [ ] actors followed by the user
- [ ] actors blocked by the user
- [ ] settings

**Comment by trymeouteh** [comment]
An account import/export feature would be cool to have which could be made for channels and those who use Peertube to follow, like and comment on videos.

Channels Import/Export
- Videos & Audios
- Video/Audio Like/Dislikes
- Video Comments
- Channel Subscribers
- Chanel Playlists
- Settings

Users
- Subscriptions
- Like/Dislikes
- Comments
- Playlists
- Settings

**Comment by trymeouteh** [comment]
I made this suggestion for channels on how Peertube could implement a automatic cloud storage backup system for channels.
https://github.com/Chocobozzz/PeerTube/issues/3082

**Comment by grravity** [comment]
Came here to suggest this, this would be really nice to have implemented.
The lack of easy account migration with federated software is a big con imo. 
I can't say I'm very worried about likes and stuff, but at minimum a built in way to simply go onto an instance as a user, say I want to migrate to this instance login as the new user, transfer the videos with all the metadata automatically, as well as my user favorites and stuff, would be amazing.

This would especially make it easier if you're on somebody elses instance, and you eventually want to migrate to your own, you can with ease. Otherwise you'd have to download each video (or upload from local) manually, reupload and retag all the metadata for each video. For large channels this is insane and very time consuming.

As others suggest, I would be all for an account export and import feature, however I very much would like to see account migration ability as well.

**Comment by othello777** [comment]
hmm, this is one of the things preventing me from jumping in right now. I feel like I have to choose the correct instance before uploading my videos because it would be a lot of work to move them manually

**Comment by Altonss** [comment]
Hello dear developers/community,
First of all thank you for this amazing piece of software! I saw that the roadmap for 2022 (and a peertube v5) has been published. Would it be possible to add this issue to the roadmap? I think that this issue is a key feature for users, to be able to export/import their data to migrate to another instance. Because of the decentralised nature of peertube, this feature is necessary. In fact some instances can shut down and users have to migrate somewhere else.

**Comment by Chocobozzz** [comment]
Hello,

We planned to work on basic account export, but we aren't sure to have time to also develop basic account import. But it's something on which we would like to move forward

**Comment by Altonss** [comment]
> We planned to work on basic account export, but we aren't sure to have time to also develop basic account import. But it's something on which we would like to move forward

That's great! Even just a basic export function would be a great improvement :)

**Comment by frankenstein91** [comment]
I need it very much now
https://blog.nikisoft.one/note/d845dd225afa7a59

**Comment by Esn024** [comment]
I strongly agree that this is an important feature. I stated my reasons on your forum [3 years ago](https://framacolibri.org/t/is-it-possible-to-backup-account-and-or-move-to-another-instance/5232):

> After all, the problem with Youtube isn’t so much that one can get banned, the problem is that moving one’s entire channel somewhere else can take many days of boring repetitive work, and the more videos you’ve done, the longer it takes, to the point where it becomes easier to just say “fine, I’ll let them censor me however they like” or “if I move I might just have to do it all again, maybe I should just stop making videos altogether”.

I just want to add that the ["exit" vs "voice"](https://en.wikipedia.org/wiki/Exit,_Voice,_and_Loyalty) framework is a good way to think of the underlying dynamics:
> members of an organization, whether a business, a nation or any other form of human grouping, have essentially two possible responses when they perceive that the organization is demonstrating a decrease in quality or benefit to the member: they can _exit_ (withdraw from the relationship); or, they can _voice_ (attempt to repair or improve the relationship through communication of the complaint, grievance or proposal for change)

At the moment, the _hosts_ have ALL the power of "voice" (in both YouTube and PeerTube) - in fact, PeerTube hosts have even more power of "voice" than YouTube, because YouTube is bigger and can sometimes be embarrassed be negative press. The only power that video creators have is thus "exit", but they have less and less "exit" power the more videos they make (because with the lack of an easy migration feature, the costs of "exit" increase more and more with every video).

**Comment by S7venLights** [comment]
I agree with all the reasons posted above and especially the previous 'exit power' argument.

But my main reason for wanting a export/migration/backup option is because I had a portfolio channel on an instance that just disappeared without warning and it took days of work to get everything back up
If there was a way of exporting and keeping followers that would be great.

**Comment by McSinyx** [comment]
If the source instance is offline then there's no way to verify
the connection between the source and target channels to notify
the followers.

**Comment by S7venLights** [comment]
everything I found on this topic:

https://github.com/Chocobozzz/PeerTube/issues/549
https://github.com/Chocobozzz/PeerTube/issues/3764
https://the-federation.info/specs/backup-restore/

https://docs.joinpeertube.org/maintain-migration
https://docs.joinpeertube.org/maintain-tools?id=create-move-video-storage-jobjs

There is the 'Transfer ownership' feature which could help as discussed [here](https://libredd.it/r/PeerTube/comments/ekmjeb/peertubevideo_one_of_the_largest_instances_is/) but that doesn't help when a instance closes without warning. We still need a local backup/export option for that.

**Comment by S7venLights** [comment]
Dammit! 3rd time I've lost a channel to PeerTube instances dying without warning and this one had hours of work!😾

https://www.orion-hub.fr/
"Due to a suspension of our dedicated server by our provider for Spam Emails, reported by UCEPROTECT and caused by our Mastodon instance, we are no longer able to offer you our PeerTube instance Orion-Hub.fr...

We are sorry for the inconvenience this causes, thank you for your understanding."

They claimed they had good funds and would keep it active always too!

PeerTube needs account migration
https://github.com/Chocobozzz/PeerTube/issues/549
Or some solution
https://github.com/Chocobozzz/PeerTube/issues/5783

**Comment by Chocobozzz** [comment]
Hi,

We plan to work on account import/export first, after PeerTube v6 scheduled for the end of 2023. Then we'll be able to continue this work to provide account migration.

**Comment by jayrope** [comment]
It seems akward, that strongly wanted features, account migration and instance migration, are laying dormant since May 2018. No particular pun intended, but **user interest is pretty clear**. What's in the way of working on this for so long? The longer this is postponed, the more difficult it gets to incorporat this into the overall development of Peertube.

**Comment by Chocobozzz** [comment]
We already prioritize features according to user interest: https://joinpeertube.org/news/roadmap-v6

Account migration is a very big feature that will need several months of work. We can't just add it to an existing roadmap where other features like chapters or replacing an existing video file is also highly requested. 

Remember that PeerTube is a free software, developed and maintained for free by only 1 developer (not full time): https://joinpeertube.org/faq#peertube-does-not-contain-all-the-tools-i-need-to-manage-my-platform

**Comment by Chocobozzz** [comment]
We plan to implement export/import feature in 2024 (not a migration tool yet, but a first step). Read the detailed blog post on https://joinpeertube.org/news/peertube-future-2024

**Comment by Chocobozzz** [comment]
Export/import has been implemented in PeerTube 6.1: https://joinpeertube.org/news/release-6.1

**Comment by S7venLights** [comment]
Great we can do some form of backup, I'm not sure what the account migration plans are but an interesting solution to maintaining a channel when domains are lost or taken down has come from odysees new solution to video hosting. Could it be integrated into PeerTube as a source of content:

Watch and read here to understand:
https://odysee.com/@Odysee:8/futureofodysee:e
&
https://odysee.com/@Odysee:8/FAQOdyseenewtech:e

(Even if your instance dies, your videos remain and can be re-linked to a new instance)
